pkg/util/namespacedname.go

```diff
@@ -0,0 +1,49 @@
+package util
+
+// same as k8s.io/apimachinery/pkg/types/namespacedname.go, but with JSON (de)serialization
+
+import (
+	"fmt"
+)
+
+// FIXME: K8s uses '/' as its separator. We currently use this for backwards-compatibility, but we
+// should change to '/'. That change will involve fixing the many places where we create strings
+// with %s:%s instead of %s/%s as well; those should just use NamespacedName directly.
+const Separator = ':'
+
+// NamespacedName represents a resource name with the namespace it's in.
+//
+// When printed with '%s' or '%v', NamespacedName is rendered as "<namespace>:<name>". Printing with
+// '%+v' or '%#v' renders as it would normally.
+type NamespacedName struct {
+	Namespace string `json:"namespace"`
+	Name      string `json:"name"`
+}
+
+func (n NamespacedName) Format(state fmt.State, verb rune) {
+	if verb == 'v' {
+		// Show fields, e.g. `{Namespace:foo Name:bar}`
+		if state.Flag('+') {
+			_, _ = state.Write([]byte(string("{Namespace:")))
+			_, _ = state.Write([]byte(n.Namespace))
+			_, _ = state.Write([]byte(string(" Name:")))
+			_, _ = state.Write([]byte(n.Name))
+			_, _ = state.Write([]byte{'}'})
+			return
+		}
+		// Go syntax representation, e.g. `util.NamespacedName{Namespace:"foo", Name:"bar"}`
+		if state.Flag('#') {
+			_, _ = state.Write([]byte(string("util.NamespacedName{Namespace:")))
+			_, _ = state.Write([]byte(fmt.Sprintf("%q", n.Namespace)))
+			_, _ = state.Write([]byte(string(", Name:")))
+			_, _ = state.Write([]byte(fmt.Sprintf("%q", n.Name)))
```

Comment:
why not combine this into a single fmt.Sprintf?

Reply:
Good point, done.